File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    data = []
    for i in range(len(priorities)):
        data.append([priorities[i], i])
    
    target = 0 
    order = 0

    while True:
        if data[target][0] == max(data, key= lambda x: x[0])[0]:
            order += 1
            if data[target][1] == location:
                return order
            data.pop(target)
        else:
            target += 1
        
        if target == len(data):
            target = 0

File: 프로그래머스/2/42587. 프로세스/프로세스.swift
import Foundation

func solution(_ priorities:[Int], _ location:Int) -> Int {
    
    var queue: [[Int]] = []
    for i in 0..<priorities.count {
        queue.append([priorities[i],i])
    }
    
    var cnt = 0
    while !queue.isEmpty {
        let firstElement = queue.map { $0[0] }
        let maxValue = firstElement.max()

        if queue[0][0] == maxValue {
            cnt += 1
            if queue[0][1] == location {
                return cnt
            }
            queue.removeFirst()
        } else {
            queue.append(queue.removeFirst())
        }
    }
    return 100
}